h/static/scripts/test/app-controller-test.coffee

```diff
@@ -156,3 +161,67 @@ describe 'AppController', ->
     fakeRoute.reload = sinon.spy()
     $scope.$broadcast(events.USER_CHANGED, {initialLoad: false})
     assert.calledOnce(fakeRoute.reload)
+
+  describe 'logout()', ->
+    it 'prompts the user if there are drafts', ->
+      callCount = window.confirm.callCount
```

Comment:
How does this even work? Where is `window.confirm` stubbed...?

The answer is [in `annotation-test.coffee`](https://github.com/hypothesis/h/blob/e970fa8bc94f75229bf7b988f67108375360521e/h/static/scripts/directive/test/annotation-test.coffee#L401), and this only works because:

- that test doesn't clean up after itself properly (it should be using `sandbox.stub` not `sinon.stub`)
- this test runs later than that one

So, in short:

- Use `$window`, not `window`, in the controller.
- Use an angular-injected fake to provide a fake `$window` object, not a global stub.
- If you don't mind, fix up the other test (perhaps in a different PR) to not leave `window.confirm` stubbed after the test has run.